src/core/qgstiledownloadmanager.cpp

```diff
@@ -198,9 +199,13 @@ QgsTileDownloadManager::QgsTileDownloadManager()
     cacheDirectory.push_back( QDir::separator() );
   }
   cacheDirectory += QLatin1String( "http-ranges" );
-  const qint64 cacheSize = settings.value( QStringLiteral( "cache/size" ), 256 * 1024 * 1024 ).toLongLong();
-
   mRangesCache->setCacheDirectory( cacheDirectory );
+  qint64 cacheSize = settings.value( QStringLiteral( "cache/size2" ), 0 ).toLongLong();
+  if ( cacheSize == 0 )
+  {
+    // Calculatre maximum cache size based on available free space
+    cacheSize = QgsNetworkDiskCache::smartCacheSize( cacheDirectory );
+  }
```

Comment:
This logic is repeated a few times -- could it be moved to a static function somewhere?

Reply:
@nyalldawson , what above moving it into the QgsNetworkDiskCache constructor?

Reply:
That'd require a qgis restart to take effect then, right?

If so, +1 so long as there's a corresponding note in the options dialog

Reply:
@nyalldawson , actually no, I think we can't. We have three similar calls, but they are distinct:
- qgsnetworkaccessmanager: we always use the cache/directory and cache/size2 settings 
- qgsserver: we use the cache/directory and cache/size2 settings unless environment variables are set
- qgstiledownloadmanager: it doesn't use QgsNetworkDiskCache at all, and uses a customized directory path derived from cache/directory

Reply:
I'll de-duplicate a bit.